c++/src/kj/build/lib.rs

```diff
@@ -0,0 +1,51 @@
+use eyre::eyre;
+use eyre::Result;
+use std::path::Path;
+
+pub fn stage_files(
+    build: &mut cc::Build,
+    files: impl Iterator<Item = impl AsRef<str>>,
+    from: impl AsRef<Path>,
+    to: impl AsRef<Path>,
+    compile: bool,
+) -> Result<()> {
+    for file in files {
+        let path = from.as_ref().join(file.as_ref());
+        println!(
+            "cargo:rerun-if-changed={}",
+            path.to_str()
+                .ok_or_else(|| eyre!("non–UTF-8 path: {:?}", path))?
+        );
+        let target = to.as_ref().join(file.as_ref());
+        std::fs::copy(&path, &target)
+            .map_err(|_| eyre!("Failed to copy {} to {}", path.display(), target.display()))?;
+        if compile {
+            build.file(target);
+        }
+    }
+
+    Ok(())
+}
+
+fn bool_int_str(p: bool) -> &'static str {
+    if p {
+        "1"
+    } else {
+        "0"
+    }
+}
+
+pub fn kj_configure<'a>(
```

Comment:
Lifetime should be elidable, not sure why clippy didn't complain

Reply:
this is because i forgot to run clippy 🙃